### PR TITLE
Support FKs in bulk

### PR DESCRIPTION
**Pull request by Deep1998**

### CODE REVIEW

File: v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/ddl/DdlTest.java [hunk 1]
```diff
@@ -322,6 +325,144 @@ public void testInterleavingPG() {
...
+   * <p>This method creates a simplified DDL with minimal schema information. Each table has only an
+   * "id" column as the primary key. Foreign key relationships are established based on the provided
+   * dependencies.
+   *
+   * @param tableNames A list of table names in the schema.
+   * @param dependencies A list of foreign key dependencies, where each dependency is represented as
+   *     a list of two strings: the child table name and the parent table name, where child
+   *     references the parent.
+   */
+  private Ddl generateDdlFromDAG(List<String> tableNames, List<List<String>> dependencies) {
```
Comment (bharadwaj-aditya) [inline v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/ddl/DdlTest.java:341]: Can you add a test for the cyclic dependency also ?

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java [hunk 2]
```diff
@@ -176,31 +235,66 @@ static PipelineResult executeShardedMigration(
 
     List<String> tablesToMigrate =
         PipelineController.listTablesToMigrate(options.getTables(), schemaMapper, ddl);
+    Set<String> tablesToMigrateSet = new HashSet<>(tablesToMigrate);
```
Comment (bharadwaj-aditya) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java:238]: Bunch of code seems duplicated - is it possible to refactor into a single method ?

**Review by bharadwaj-aditya — COMMENTED** [review]
LGTM overall . Pleaseadd a test for cyclic checks in ddl.

File: v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/ddl/DdlTest.java (hunk 1 above)
Comment (Deep1998) [reply v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/ddl/DdlTest.java:341]: Its already there at the end

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java (hunk 2 above)
Comment (Deep1998) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java:238]: Skipping as per offline discussion

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Pull request merged**